packages/contracts-periphery/contracts/foundry-tests/citizen-house/BadgeAdmin.t.sol

```diff
@@ -0,0 +1,692 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.15;
+
+import { BadgeAdmin } from "../../universal/citizen-house/BadgeAdmin.sol";
+import { Badge } from "../../universal/citizen-house/Badge.sol";
+import { Test, stdError } from "forge-std/Test.sol";
+
+function mkadr(string memory name) pure returns (address) {
+    address adr = address(uint160(uint256(keccak256(abi.encodePacked(name)))));
+    return adr;
+}
+
+contract TBadge is Badge {
+    constructor() Badge("example", "ex", "example.com") {}
+}
+
+contract TBadgeAdmin is BadgeAdmin {
+    TBadge internal badge;
+
+    address[] opAdrs = [mkadr("op")];
+    address[] adrs = [0xa8B3478A436e8B909B5E9636090F2B15f9B311e7];
+
+    function setUp() public {
+        badge = new TBadge();
+    }
+
+    constructor() BadgeAdmin(address(badge), 100, 100, 100, opAdrs) {}
+}
+
+contract BadgeAdminTest is Test {
+    TBadgeAdmin internal badgeAdmin;
+    TBadge internal badge;
+
+    address opAdr = mkadr("op"); //[0xa8B3478A436e8B909B5E9636090F2B15f9B311e7];
+    bytes32 testIPFSHash = 0x0170171c23281b16a3c58934162488ad6d039df686eca806f21eba0cebd03486;
+
+    function setUp() public {
+        badgeAdmin = new TBadgeAdmin();
+        badge = new TBadge();
+    }
+
+    function getOPCOSet(uint256 set) public pure returns (address[] memory) {
+        address[] memory opcoAdrs = new address[](4);
+        if (set == 0) {
+            opcoAdrs[0] = mkadr("opco1");
+            opcoAdrs[1] = mkadr("opco2");
+            opcoAdrs[2] = mkadr("opco3");
+            opcoAdrs[3] = mkadr("opco4");
+        } else if (set == 1) {
+            opcoAdrs[0] = mkadr("opco5");
+            opcoAdrs[1] = mkadr("opco6");
+            opcoAdrs[2] = mkadr("opco7");
+            opcoAdrs[3] = mkadr("opco8");
+        } else if (set == 2) {
+            opcoAdrs[0] = mkadr("opco9");
+            opcoAdrs[1] = mkadr("opco10");
+            opcoAdrs[2] = mkadr("opco11");
+            opcoAdrs[3] = mkadr("opco12");
+        } else if (set == 3) {
+            opcoAdrs[0] = mkadr("opco13");
+            opcoAdrs[1] = mkadr("opco14");
+            opcoAdrs[2] = mkadr("opco15");
+            opcoAdrs[3] = mkadr("opco16");
+        } else if (set == 4) {
+            opcoAdrs[0] = mkadr("opco17");
+            opcoAdrs[1] = mkadr("opco18");
+            opcoAdrs[2] = mkadr("opco19");
+            opcoAdrs[3] = mkadr("opco20");
+        } else if (set == 5) {
+            opcoAdrs[0] = mkadr("opco21");
+            opcoAdrs[1] = mkadr("opco22");
+            opcoAdrs[2] = mkadr("opco23");
+            opcoAdrs[3] = mkadr("opco24");
+        }
+
+        return opcoAdrs;
+    }
+
+    function getCitizenSet(uint256 set) public pure returns (address[] memory) {
+        address[] memory citizenAdrs = new address[](4);
+        if (set == 0) {
```

Comment:
Is there a particular pattern to why the test data is set up this way? Its unclear what is being asserted on because this data is retrieved far from where its being constructed, so i need to go back and forth to review

Reply:
This question applies more to the supply sets

Reply:
No particular reason - refactored to be more specific!